internal/services/stock_exchange/algorithms.go

```diff
@@ -0,0 +1,289 @@
+package stock_exchange
+
+import (
+	"Tradeasy/config"
+	model "Tradeasy/internal/model/stock_exchange"
+	"math"
+	"time"
+)
+
+// BuyOrder ...Update Buy Order actions on the StockExchange database
+func BuyOrder(buyOrderBody OrderRequest) (resp OrderResponse, err error) {
+
+	orderId := buyOrderBody.OrderID
+	quantity := buyOrderBody.Quantity
+	stock := buyOrderBody.StockName
+	price := buyOrderBody.LimitPrice
+	orderType := buyOrderBody.OrderType
+
+	resp.Status = "Cancelled by engine"
+	resp.OrderID = orderId
+	resp.StockName = stock
+	resp.Message = "Couldn't execute due to unavailability of shares"
+
+	var sellBook []model.SellOrderBook
+	// db lock
+	err = config.DB.Raw("SELECT * FROM sell_order_book WHERE stock_ticker_symbol = ?  ORDER BY order_price ASC,created_at ASC ", stock).Scan(sellBook).Error
```

Comment:
aren't we using gorm functionality?

Reply:
added it few places, everywhere might take a lot of time